server/src/main/java/io/crate/role/DropRoleTask.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.role;
+
+import java.util.Collection;
+
+import org.elasticsearch.cluster.AckedClusterStateUpdateTask;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.common.Priority;
+
+import io.crate.common.annotations.VisibleForTesting;
+import io.crate.role.metadata.RolesMetadata;
+import io.crate.role.metadata.UsersMetadata;
+import io.crate.role.metadata.UsersPrivilegesMetadata;
+
+public class DropRoleTask extends AckedClusterStateUpdateTask<WriteRoleResponse> {
+
+    private final DropRoleRequest request;
+    private boolean alreadyExists = true;
+
+    DropRoleTask(DropRoleRequest request) {
+        super(Priority.URGENT, request);
+        this.request = request;
+    }
+
+    @Override
+    public ClusterState execute(ClusterState currentState) throws Exception {
+        Metadata currentMetadata = currentState.metadata();
+        Metadata.Builder mdBuilder = Metadata.builder(currentMetadata);
+        alreadyExists = dropRole(
+            mdBuilder,
+            request.roleName()
+        );
+        return ClusterState.builder(currentState).metadata(mdBuilder).build();
```

Comment:
`alreadyExists` doesn't make sense to me in the context of dropping a role. It should probably be `alreadyMissing`, or `notFound` or something like that?

And if it was a no-op it would be good to return `currentState` as is, so that the component applying cluster state changes is aware that there was no change and this doesn't need to propagate through the cluster.